proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
For what it's worth, I hold the opinion that 'query parameters' is a convenience that's not needed at all in the base spec, because the same information can be extracted from getUri().

If I can't have that, I would suggest to only provide a getter that *always* reflects what's in the uri.

If a setter is a must, I think the result of getUri() and getQueryParams() *should* always be consistent.